src/Test/ImplicitMethodsIntegrationTest.php

```diff
@@ -0,0 +1,226 @@
+        $middleware2 = $this->prophesize(MiddlewareInterface::class)->reveal();
+        $route1 = new Route('/api/v1/me', $middleware1, [RequestMethod::METHOD_GET]);
+        $route2 = new Route('/api/v1/me', $middleware2, [RequestMethod::METHOD_POST]);
+
+        $router = $this->getRouter();
+        $router->addRoute($route1);
+        $router->addRoute($route2);
+
+        $finalResponse = (new Response())->withHeader('foo-bar', 'baz');
+        $finalResponse->getBody()->write('FOO BAR BODY');
+
+        $finalHandler = $this->prophesize(RequestHandlerInterface::class);
+        $finalHandler
+            ->handle(Argument::that(function (ServerRequestInterface $request) use ($route1) {
+                if ($request->getMethod() !== RequestMethod::METHOD_GET) {
+                    return false;
+                }
+
+                if ($request->getAttribute(ImplicitHeadMiddleware::FORWARDED_HTTP_METHOD_ATTRIBUTE)
+                    !== RequestMethod::METHOD_HEAD
+                ) {
+                    return false;
+                }
+
+                $routeResult = $request->getAttribute(RouteResult::class);
+                if (! $routeResult) {
+                    return false;
+                }
+
+                if (! $routeResult->isSuccess()) {
+                    return false;
+                }
+
+                $matchedRoute = $routeResult->getMatchedRoute();
+                if (! $matchedRoute) {
+                    return false;
+                }
+
+                if ($matchedRoute !== $route1) {
+                    return false;
+                }
+
+                return true;
+            }))
+            ->willReturn($finalResponse)
+            ->shouldBeCalledTimes(1);
+
+        $routeMiddleware = new PathBasedRoutingMiddleware($router);
+        $handler = new TestAsset\ImplicitHeadHandler($finalHandler->reveal());
+        $request = new ServerRequest([], [], '/api/v1/me', RequestMethod::METHOD_HEAD);
+
+        $response = $routeMiddleware->process($request, $handler);
+
+        $this->assertEquals(StatusCode::STATUS_OK, $response->getStatusCode());
+        $this->assertEmpty((string) $response->getBody());
+        $this->assertTrue($response->hasHeader('foo-bar'));
+        $this->assertSame('baz', $response->getHeaderLine('foo-bar'));
+    }
+
+    public function testImplicitOptionsRequest()
+    {
+        $middleware1 = $this->prophesize(MiddlewareInterface::class)->reveal();
+        $middleware2 = $this->prophesize(MiddlewareInterface::class)->reveal();
+        $route1 = new Route('/api/v1/me', $middleware1, [RequestMethod::METHOD_GET]);
+        $route2 = new Route('/api/v1/me', $middleware2, [RequestMethod::METHOD_POST]);
+
+        $router = $this->getRouter();
+        $router->addRoute($route1);
+        $router->addRoute($route2);
+
+        $finalHandler = $this->prophesize(RequestHandlerInterface::class);
+        $finalHandler->handle()->shouldNotBeCalled();
```

Comment:
We should use here const `Webimpress\HttpMiddlewareCompatibility\HANDLER_METHOD`.